toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -239,10 +239,56 @@ func CreateSparseDisk(diskPath string, size uint64, perm os.FileMode) (err error
 	return
 }
 
+// devineBlockSize tries to determine the block size used by the file system image
+func devineBlockSize(diskFilePath string) uint32 {
+	var blockSize uint32 = 512 // Default to 512
+	diskFile, err := os.Open(diskFilePath)
+	if err != nil {
+		logger.Log.Warnf("Failed to open %v to devine block size, assuming %d.", diskFilePath, blockSize)
+		return blockSize
+	}
+	defer diskFile.Close()
+
+	var probePoint int64 = 4096
+	signature := make([]byte, 8)
+	_, err = diskFile.ReadAt(signature, probePoint)
+	if err != nil {
+		logger.Log.Warnf("Failed to read from %v to devine block size, assuming %d.", diskFilePath, blockSize)
+		return blockSize
+	}
+	// GPT header starts with 8 bytes of ascii "EFI PART"
+	if signature[0] == 0x45 &&
```

Comment:
Use `bytes.Equal`

Reply:
OK. Lack of knowledge of go is surely biting me here.